src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -509,6 +520,48 @@ ErrorIfUnsupportedCreateCitusLocalTable(Relation relation)
 }
 
 
+/*
+ * EnsureIfFdwHasTableName errors out with a hint if the foreign table is using postgres_fdw and
+ * the option table_name is not provided.
+ */
+static void
+EnsureIfFdwHasTableName(Oid relationId)
+{
+	char relationKind = get_rel_relkind(relationId);
+	if (relationKind == RELKIND_FOREIGN_TABLE)
```

Comment:
should/can we do this only for postgres_fdw? I believe the code used to only check for postgres_fdw, but I cannot see we filter that here (maybe I'm missing something)?

Reply:
I think that's a valid case for any other extension that supports foreign tables. So I wouldn't say we should add a check for only postgres_fdw.

Reply:
I was wrong :) After we discussed this with Onder in private we agreed to make this check only for postgres_fdw.